remote-wallet/src/remote_wallet.rs

```diff
@@ -93,15 +120,20 @@ pub struct RemoteWalletManager {
     #[cfg(feature = "hidapi")]
     usb: Arc<Mutex<hidapi::HidApi>>,
     devices: RwLock<Vec<Device>>,
+    trezor_client: Option<Rc<RefCell<Trezor>>>,
 }
 
 impl RemoteWalletManager {
     /// Create a new instance.
     #[cfg(feature = "hidapi")]
-    pub fn new(usb: Arc<Mutex<hidapi::HidApi>>) -> Rc<Self> {
+    pub fn new(
+        usb: Arc<Mutex<hidapi::HidApi>>,
+        trezor_client: Option<Rc<RefCell<Trezor>>>,
```

Comment:
can we not populate this based on the contents of the hid device list? better to not need implementation details directly exposed to the api

Reply:
I don't think we can. Trezor uses Protobuf v2 for host-device communication and as a consequence the `trezor-client` crate has a `WebUsbTransport` transport implementation that utilizes the `rusb` crate for USB communication with Trezor devices, and a `UdpTransport` implementation that uses standard Rust networking libraries for communication over UDP. It seems to me if we wanted to use the `hidapi` crate it would involve integrating the `hidapi` crate into the `trezor-client` transport module which could require significant changes to the codebase depending on how tightly coupled the existing transport implementation is with the rest of the crate.